python/pyspark/worker.py

```diff
@@ -628,9 +628,36 @@ def verify_result(result):
                 )
                 return result
 
-            return lambda *a, **kw: map(
-                lambda res: (res, arrow_return_type), map(verify_result, f(*a, **kw))
-            )
+            def evaluate(*args: pd.Series, **kwargs: pd.Series):
+                try:
+                    if len(args) == 0 and len(kwargs) == 0:
+                        yield verify_result(pd.DataFrame(f())), arrow_return_type
```

Comment:
I see we have a try catch statement for the entire evaluate function.  Can we actually only wrap the invocation of `f()`? This is to separate the user's code error from the error thrown by the worker code.